server/server.go

```diff
@@ -25,11 +29,34 @@ func New() *Backend {
 func (b *Backend) AddUser(ctx context.Context, user *pbExample.User) (*types.Empty, error) {
 	b.mu.Lock()
 	defer b.mu.Unlock()
+
+	if len(b.users) == 0 && user.GetRole() != pbExample.Role_ADMIN {
+		st := status.New(codes.InvalidArgument, "first user created must be an admin")
+		// Note that st.WithDetails requires a proto.Message that has been registered
+		// with golang/protobuf to work. This in turn requires us to instrument our
```

Comment:
Does it help if we create a gogo/googleapis/errdetails and then manually register that to golang/protobuf/proto?

Reply:
I hope that adding that one line to the example is maybe less effort and hopefully helps to remove a lot of other dependencies?

I don't know how to decide for users if they want to import golang/protobuf/proto and gogo/protobuf/proto or just gogo/protobuf/proto :(

Maybe we can make a seperate repo that requires github.com/gogo/googleapis and just does the registering to golang/protobuf/proto for you?  But then I would first want to know whether doing it manually works for this case.

Reply:
I hope registering also solves
https://github.com/grpc-ecosystem/grpc-gateway/blob/58f78b988bc393694cef62b92c5cde77e4742ff5/runtime/query.go#L275
That was brought up by Timon Wong on your blog.
But I am unsure.

Reply:
I'm happy to keep this as it is as well. I definitely understand the concern about making it some sort of standard practice to register things with both gogo/protobuf and golang/protobuf. We've got a working workaround here, and it's not ideal, but it's certianly not the fault of gogo/protobuf, and it's only a problem with the way gRPC-Gateway uses the grpc Status package. I could perhaps consider making a contribution to the gatway to make it independent of the google/protobuf/ptypes package.

Reply:
That would be commendable.
But I guess this way also works.

Could you add a test for field mask, to make sure it also works?

Reply:
Ah good idea, yes.

Reply:
Actually I think the fiedmask thing is more suited to a separate issue and PR. I'm not even sure what it's used for, but I think we should have a separate method that takes a FieldMask as a (query?) parameter as that code appears to be specific to that. I think this PR is good to go in if you're happy with it in its current state.

Reply:
Cool ok lets log the FieldMask as a separate issue to tackle later.